File: js/remove-nth-node-end-list/remove-nth-node-end-list.js
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

function removeNthFromEnd(head, n){
    let slow = head, fast = head
    for (let i = 0; i < n; i++) fast = fast.next
    if (!fast) return head.next
    while (fast.next) {
        fast = fast.next
        slow = slow.next
    }
    slow.next = slow.next.next
    return head
}

let node5 = new ListNode(5)
let node4 = new ListNode(4, node5)
let node3 = new ListNode(3, node4)
let node2 = new ListNode(2, node3)
let node1 = new ListNode(1, node2)

let res = removeNthFromEnd(node1, 2)

console.log(`answer to the first example is: [${res.val}, ${res.next.val}, ${res.next.next.val}, ${res.next.next.next.val}]`)

File: go/remove-nth-node-end-list/remove-nth-node-end-list.go
package main

import(
	"fmt"
) 

type ListNode struct {
    Val int
    Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
    slow, fast := head, head
    for i := 0; i < n; i++ {
        fast = fast.Next
    }
    if fast == nil {
        return head.Next
    }
    for fast.Next != nil {
        fast = fast.Next
        slow = slow.Next
    }
    slow.Next = slow.Next.Next
    return head
}

func main(){
	node1 := &ListNode{Val: 1}
	node2 := &ListNode{Val: 2}
	node3 := &ListNode{Val: 3}
	node4 := &ListNode{Val: 4}
	node5 := &ListNode{Val: 5}

	node1.Next = node2
	node2.Next = node3
	node3.Next = node4
	node4.Next = node5

    res := removeNthFromEnd(node1, 2)


	fmt.Printf("answer to the first example is: [%v, %v, %v, %v]", res.Val, res.Next.Val, res.Next.Next.Val, res.Next.Next.Next.Val)
}